### ISSUE TITLE
Update and refactor knative-serving charts

### ISSUE THREAD

**Issue by aerfio** [kyma-project/kyma#7854]
**Description**
Knative serving got now release to `v0.13`, which brings some changes to our internal knative-serving chart. While doing update please recheck that everything is ok with it and **please separate each and every** resource into it's own file (so no `serving.yaml` file that spans 1000+ lines....).

Please make sure that this chart is robust as hell, to prevent issues with it randomly not installing on CI

**Reasons**
We need to be responsible for our charts and currently no one knows what the hell is going on there. Update and refactor might make it more stable and easier to work with.

**Attachments**

**Comment by michal-hudy** [comment]
I look like we need to upgrade serving before moving to the solution based on pure Kubernetes resources. Knative Serving is stuck really often with status `RevisionMissing`.

**Comment by magicmatatjahu** [comment]
I made investigation and found some problems on bumping from `0.12.1` to `0.14.0` version of KNative Serving:
- from `0.13.X` Knative has deprecation of the Alpha and Beta APIs. On `0.13.X` it's only deprecated, but on `0.14.0` KNative had new default storage version (from `v1alpha1` to `v1`) and also had migration job for CRD for that, but on https://github.com/kyma-project/kyma/pull/8508 PR I didn't change this storage version and default using of Service works as aspected (creating/running/updating). Latest version also contains conversion webhooks for `routes`, `configurations`, `services`, `revisions` to `v1`, running in `webhook` (`knative-serving` release) deployment. We have a two helm releases `knative-serving` and `knative-serving-init` and this is a problem, because K8s cannot see this webhook on installation (not upgrade), because init is before the real release.
- some related resources is focused on `v1` APIs (eg https://github.com/knative/serving/pull/6957/files) and I don't know what it has an impact for e.g. our serverless. We still use there `v1alpha1` API.

I changed images for controllers, add new `istio-webhook` (CRs for that) for adding canonical istio labels -> https://github.com/knative/net-istio/commit/4118da04bbf82d8fa022bef8c94f23f36a3d0b0f and serverless works as aspected in simple flow, but because of problems with images from `quay.io` I can't test all cases and I don't know if everything works.
